smartconfig/file.py

```diff
@@ -0,0 +1,94 @@
+import yaml
+from os import PathLike
+from typing import Mapping, MutableMapping, Union
+
+from smartconfig import _registry
+from smartconfig.constructors import _ref_constructor
+
+
+def _update_mapping(source: Mapping, dest: MutableMapping, path: str = "") -> MutableMapping:
```

Comment:
This doesn't account for a weird situation where the YAML has a non-mapping node stored as the value to the key corresponding to a `ConfigEntry`. It's tricky to check because we don't know all possible `ConfigEntry` paths ahead of time. The solution may involve both checking against known entries in this function and checking the current value of the path in `_register_entry`.

Reply:
Now enforced in 120349f.